07-map.slide

```diff
@@ -2,19 +2,148 @@ Maps in Go
 22 Feb 2017
 Tags: edmontongo, golang, workshop
 
+Lukasz Rozycki
+[[https://lukaszr.com]]
+
 * Maps
 
-composite literal for creating maps
+- Hash Table, Dictionary
+- Mapping from key to a value
+- Fast lookups, adds, and deletes
+- A Go map type looks like this:
+ map[KeyType]ValueType
+- Maps are not safe for concurrent use
+
+* What can be a key?
+
+- Any type that is comparable
+ ==    equal
+ !=    not equal
+- boolean, numeric, string, pointer
+- interface type, channel
+- struct or array containing above
+- Not slice, map, nor function
+
+* Creating a map 
+- composite literal (note trailing comma)
+.play 07-map/compositeliteral.go /START/,/END/
+- empty map
+.play 07-map/emptyliteral.go /START/,/END/
+
+* Lookup
+- index expression
+.play 07-map/lookup.go /START/,/END/
+- zero value
+.play 07-map/zerovalue.go /START/,/END/
+- what about Baby?
+.play 07-map/commaok1.go /START/,/END/
+
+* Lookup
+- "comma ok" idiom
+.play 07-map/commaok2.go /START/,/END/
+- concise notation
+.play 07-map/commaok3.go /START/,/END/
+
+* Lookup
+- ignore age
+.play 07-map/commaok4.go /START/,/END/
+- why underscore?
```

Comment:
oops, I thought I'd be the first one to mention underscore. heh.

oh well. reinforcement is always good